.buildkite/hooks/post-checkout

```diff
@@ -32,6 +34,49 @@ checkout_merge() {
     }
 }
 
+retry() {
+    local retries=$1
+    shift
+
+    local count=0
+    until "$@"; do
+        exit=$?
+        wait=$((2 ** count))
+        count=$((count + 1))
+        if [ $count -lt "$retries" ]; then
+            >&2 echo "Retry $count/$retries exited $exit, retrying in $wait seconds..."
+            sleep $wait
+        else
+            >&2 echo "Retry $count/$retries exited $exit, no more retries left."
+            return $exit
+        fi
+    done
+    return 0
+}
+
+GITHUB_REPO_TOKEN=$(retry 5 vault kv get -field token ${GITHUB_TOKEN_VAULT_PATH})
```

Comment:
Usually secrets are retrieved in the `pre-command` hook. Why are those needed here?

Reply:
The answers here: 
https://github.com/elastic/fleet-server/pull/2702#discussion_r1239715942 
and here:
https://github.com/elastic/fleet-server/pull/2702#discussion_r1239721118